cpp/daal/src/algorithms/linear_model/linear_model_predict_dense_default_batch_impl.i

```diff
@@ -88,6 +89,50 @@ void PredictKernel<algorithmFPType, defaultDense, cpu>::computeBlockOfResponses(
     } /* if (findBeta0) */
 } /* void PredictKernel<algorithmFPType, defaultDense, cpu>::computeBlockOfResponses */
 
+template <typename algorithmFPType, CpuType cpu>
+void PredictKernel<algorithmFPType, defaultDense, cpu>::computeBlockOfResponsesSOA(const size_t & startRow, DAAL_INT * numFeatures,
+                                                                                   DAAL_INT * numRows, NumericTable * dataTable, DAAL_INT * numBetas,
+                                                                                   const algorithmFPType * beta, DAAL_INT * numResponses,
+                                                                                   algorithmFPType * responseBlock, bool findBeta0,
+                                                                                   DAAL_INT * numRowsInData)
+{
+    SafeStatus safeStat;
+    algorithmFPType one       = 1.0;
+    const DAAL_INT nFeatures  = *numFeatures;
+    const DAAL_INT nRows      = *numRows;
+    const DAAL_INT nResponses = *numResponses;
+    char trans                = 'T';
+    char notrans              = 'N';
+
+    services::internal::service_memset_seq<algorithmFPType, cpu>(responseBlock, algorithmFPType(0.0), nRows * nResponses);
+
+    size_t numBlocks = nFeatures / _numRowsInBlock;
+    numBlocks += numBlocks * _numRowsInBlock < nFeatures;
+    for (size_t iBlock = 0; iBlock < numBlocks; ++iBlock)
+    {
+        size_t startCol = iBlock * _numRowsInBlock;
+        size_t endCol   = (iBlock == numBlocks - 1 ? nFeatures : startCol + _numRowsInBlock);
+
+        DAAL_INT colSize = endCol - startCol;
+
+        ReadColumns<algorithmFPType, cpu> xBlock(dataTable, startCol, startRow, nRows);
+        DAAL_CHECK_BLOCK_STATUS_THR(xBlock);
+        const algorithmFPType * data = xBlock.get();
```

Comment:
we did this on the assumption of data continuity. we need to check this before executing this code, or copy otherwise.

use method `isHomogeneousFloatOrDouble` from SOA table